lessons/06-tool-calling/README.md

```diff
@@ -0,0 +1,426 @@
+# Lesson 6: Tool calling
+
+Tool calling, or function calling that it's also known as is about providing capabilities to your AI model it didn't have before. The idea is to provide it with meta descriptions of your actual functions and make the AI model point out when such a tool should be called given a user's prompt.
```

Comment:
This is missing a word or two it seems. Here's another way to phrase it:

Tool calling, also known as function calling, refers to enhancing your AI model by adding capabilities it previously lacked. The concept involves supplying meta descriptions of your functions, allowing the AI model to determine when a particular tool should be called based on a user's prompt.